keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ResourceSpec.kt

```diff
@@ -30,4 +29,14 @@ interface ResourceSpec {
    * other fields.
    */
   val displayName: String
+
+  /**
+   * Applies the given [suffix] to the resource [id], and to all aggregate properties of the spec
+   * whose names are derived from the [id].
+   *
+   * @return a copy of the original [ResourceSpec] with the modified identifiers.
+   */
+  @JvmDefault
+  fun deepRename(suffix: String): ResourceSpec =
+    throw UnsupportedOperationException("Not implemented")
```

Comment:
Intentional unsupported I'd guess for now?

Reply:
Kotlin nit: should use `TODO()`

Reply:
Found this `deepRename` call is used for [Preview / PR environments](https://github.com/osoriano/keel/blob/port-netflix-changes/keel-scm/src/main/kotlin/com/netflix/spinnaker/keel/preview/PreviewEnvironmentCodeEventListener.kt#L324). I haven't used this feature yet though

I was tempted to remove the default `UnsupportedOperationException` here, but it could cause plugins to break if it's not implemented, so I moved to `TODO()` :+1: